lib/src/workspace.rs

```diff
@@ -194,6 +196,17 @@ impl Workspace {
         workspace_root: &Path,
         git_repo_path: &Path,
     ) -> Result<(Self, Arc<ReadonlyRepo>), WorkspaceInitError> {
+        let mut git_repo_path = git_repo_path
+            .canonicalize()
+            .map_err(|_| WorkspaceInitError::PathNotFound(git_repo_path.to_path_buf()))?;
+        if !git_repo_path.ends_with(".git") {
+            git_repo_path.push(".git");
+
+            if !git_repo_path.exists() {
+                git_repo_path.pop();
+            }
+        }
+
```

Comment:
I'm not strongly against this change, but I think ".git" should be appended by CLI code. `git_repo_path` here should point to the git repository, not the workspace.

Reply:
I think you're right. I saw the canonicalization below and figured it was consistent to normalize the `git_repo_path` here too, but I missed the TODO saying that we planned to move the canonicalization below out of this function.

Reply:
Circling back to clean things up.

I think I've managed to get myself confused by the terms. Help check my understanding:

Workspace path - This is the directory the user works in. It contains the 'jj' repo and **may** also contain a 'git' repo.

Repo path - This holds the "special" folder for git/jj, so for jj, this is the path to the .jj directory, but for git, it's the path to the .git directory.

Is that correct?

What should `git_repo_path` point to then? 

I guess, technically, we want it to point to the '.git' directory, but for the user, it's a bit annoying UI wise, so we can accept the parent directory and normalize it ourselves.

Am I understanding this correctly?

Thanks! :-D

Reply:
> Workspace path - This is the directory the user works in. It contains the 'jj' repo and **may** also contain a 'git' repo.
> 
> Repo path - This holds the "special" folder for git/jj, so for jj, this is the path to the .jj directory, but for git, it's the path to the .git directory.

Basically yes, that's my understanding. For `jj`, the repo path is probably `.jj/repo`. For "bare" git repo, the path might not end with ".git".

> What should `git_repo_path` point to then?
> 
> I guess, technically, we want it to point to the '.git' directory, but for the user, it's a bit annoying UI wise, so we can accept the parent directory and normalize it ourselves.

Yes. That's why I insist that it's CLI/UI business to append ".git".

FWIW, we call `-R/--repostiroy` a repository path in CLI, but it's actually a workspace path.